### ISSUE TITLE
Error variants named `None` conflict with `Option::None`

### ISSUE THREAD

**Issue by Enet4** [shepmaster/snafu#51]
If by chance someone chooses to name one of the variants `None`, the code expanded by `derive(Snafu)` will not compile.

```rust
use snafu::Snafu;

#[derive(Debug, Snafu)]
enum Error {
    #[snafu(display("Nothing is wrong"))]
    None,
}
```

```none
error[E0308]: mismatched types
 --> src/main.rs:3:17
  |
3 | #[derive(Debug, Snafu)]
  |                 ^^^^^ expected enum `std::option::Option`, found struct `None`
  |
  = note: expected type `std::option::Option<&dyn std::error::Error>`
             found type `None`

error[E0308]: mismatched types
 --> src/main.rs:3:17
  |
3 | #[derive(Debug, Snafu)]
  |                 ^^^^^ expected enum `std::option::Option`, found struct `None`
  |
  = note: expected type `std::option::Option<&(dyn std::error::Error + 'static)>`
             found type `None`

error[E0308]: mismatched types
 --> src/main.rs:3:17
  |
3 | #[derive(Debug, Snafu)]
  |                 ^^^^^ expected enum `std::option::Option`, found struct `None`
  |
  = note: expected type `std::option::Option<&snafu::backtrace_shim::Backtrace>`
             found type `None`

error: aborting due to 3 previous errors
```

**Comment by shepmaster** [comment]
Ugh, I was so good about fully-qualifying other things, but `Some` and `None` are just so ingrained....

**Comment by shepmaster** [comment]
Released in 0.2.2
